### PR TITLE
Continue as new

### LINKED ISSUE #141: [Feature Request] Implement ContinueAsNew
Once it is supported by sdk-core

### PR DESCRIPTION
**Pull request by bergundy**
Closes #141

There are still some TODOs in the code to document `memo` and `searchAttributes`

### CODE REVIEW

File: packages/meta/README.md [hunk 1]
```diff
@@ -18,13 +18,16 @@ Learn how to use Temporal on the [docs site](https://docs.temporal.io/docs/node/
 
 #### Features
 
-Partial implementation of all components that make up a Temporal SDK - Worker, Client, Workflows, and Activities
+Mostly complete implementation of all components that make up a Temporal SDK - Worker, Client, Workflows, and Activities
```
Comment (Sushisource) [inline packages/meta/README.md:21]: Lol, love this change :smile:

File: packages/test/src/test-workflows.ts [hunk 2]
```diff
@@ -1518,3 +1518,25 @@ test('log-before-timing-out', async (t) => {
...
+            taskQueue: 'test',
+            arguments: defaultDataConverter.toPayloads('signal'),
+          },
+        },
+      ])
+    );
+  }
+});
+
+test.todo('no-commands-can-be-issued-once-workflow-completes');
```
Comment (Sushisource) [inline packages/test/src/test-workflows.ts:1542]: This one needs to be filled out?

File: packages/workflow/src/interceptors.ts [hunk 3]
```diff
@@ -105,15 +112,13 @@ export interface WorkflowOutboundCallsInterceptor {
...
-    next: Next<WorkflowOutboundCallsInterceptor, 'scheduleActivity'>
-  ) => Promise<unknown>;
+  scheduleActivity?: (input: ActivityInput, next: Next<this, 'scheduleActivity'>) => Promise<unknown>;
 
   /**
    * Called when Workflow starts a timer
    */
-  startTimer?: (input: TimerInput, next: Next<WorkflowOutboundCallsInterceptor, 'startTimer'>) => Promise<void>;
+  startTimer?: (input: TimerInput, next: Next<this, 'startTimer'>) => Promise<void>;
+  continueAsNew?: (input: ContinueAsNewInput, next: Next<this, 'continueAsNew'>) => Promise<never>;
```
Comment (Sushisource) [inline packages/workflow/src/interceptors.ts:121]: Needs a simple docstring

**Review by Sushisource — APPROVED** [review]
LGTM just nits

File: packages/test/src/test-workflows.ts (hunk 2 above)
Comment (bergundy) [reply packages/test/src/test-workflows.ts:1542]: Not strictly related to this PR but I felt like it at least deserves a placeholder

File: packages/workflow/src/interceptors.ts (hunk 3 above)
Comment (bergundy) [reply packages/workflow/src/interceptors.ts:121]: 👍

**Pull request merged**